accumulo/src/main/scala/geotrellis/spark/io/accumulo/AccumuloLayerCopier.scala

```diff
@@ -1,55 +1,120 @@
-    new SparkLayerCopier[AccumuloLayerHeader, K, V, M](
-      attributeStore = AccumuloAttributeStore(instance.connector),
-      layerReader    = layerReader,
-      layerWriter    = layerWriter
-    ) {
-      def headerUpdate(id: LayerId, header: AccumuloLayerHeader): AccumuloLayerHeader = {
-        val writerHeader = writerAttributeStore.readLayerAttribute[AccumuloLayerHeader](id, Fields.header)
-        header.copy(tileTable = writerHeader.tileTable)
-      }
+class AccumuloLayerCopier(
+  attributeStore: AttributeStore[JsonFormat],
+  layerReader: AccumuloLayerReader,
+  getLayerWriter: LayerId => AccumuloLayerWriter
+) extends LayerCopier[LayerId] {
+  def copy[
+    K: AvroRecordCodec: Boundable: JsonFormat: ClassTag,
+    V: AvroRecordCodec: ClassTag,
+    M: JsonFormat: Component[?, Bounds[K]]
+  ](from: LayerId, to: LayerId): Unit = {
+    if (!attributeStore.layerExists(from)) throw new LayerNotFoundError(from)
+    if (attributeStore.layerExists(to)) throw new LayerExistsError(to)
+
+    val (_, _, keyIndex, _) = try {
+      attributeStore.readLayerAttributes[AccumuloLayerHeader, M, KeyIndex[K], Schema](from)
+    } catch {
+      case e: AttributeNotFoundError => throw new LayerCopyError(from, to).initCause(e)
+    }
+
+    try {
+      getLayerWriter(from).write(to, layerReader.read[K, V, M](from), keyIndex)
+    } catch {
+      case e: Exception => new LayerCopyError(from, to).initCause(e)
     }
   }
+}
+
+object AccumuloLayerCopier {
+  def apply(
+    attributeStore: AttributeStore[JsonFormat],
+    layerReader: AccumuloLayerReader,
+    getLayerWriter: LayerId => AccumuloLayerWriter
+  )(implicit sc: SparkContext): AccumuloLayerCopier =
+    new AccumuloLayerCopier(
+      attributeStore,
+      layerReader,
+      getLayerWriter
+    )
 
-  def apply[K: Boundable: JsonFormat: ClassTag, V: ClassTag, M: JsonFormat](
+  def apply(
     attributeStore: AttributeStore[JsonFormat],
-    layerReader: AccumuloLayerReader[K, V, M],
-    layerWriter: AccumuloLayerWriter[K, V, M]
-  )(implicit sc: SparkContext): SparkLayerCopier[AccumuloLayerHeader, K, V, M] =
-    apply[K, V, M](
-      attributeStore = attributeStore,
-      layerReader    = layerReader,
-      layerWriter    = layerWriter
+    layerReader: AccumuloLayerReader,
+    layerWriter: AccumuloLayerWriter
+  )(implicit sc: SparkContext): AccumuloLayerCopier =
+    apply(
+      attributeStore,
+      layerReader,
+      _ => layerWriter
+    )
+
+  def apply(
+    instance   : AccumuloInstance,
+    layerReader: AccumuloLayerReader,
+    layerWriter: AccumuloLayerWriter
+  )(implicit sc: SparkContext): AccumuloLayerCopier =
+    apply(
+      AccumuloAttributeStore(instance.connector),
+      layerReader,
+      _ => layerWriter
     )
 
-  def apply[K: Boundable: AvroRecordCodec: JsonFormat: ClassTag, V: AvroRecordCodec: ClassTag, M: JsonFormat](
+  def apply(
+    instance: AccumuloInstance,
+    table: String,
```

Comment:
While it's the only logical choice, since we must know the table which we copy from I think it would help the API readability to rename all these `table` arguments to `targetTable`.